code/controllers/subsystem/movement/zfall.dm

```diff
@@ -0,0 +1,36 @@
+SUBSYSTEM_DEF(zfall)
+	name = "Zfall"
+	flags = SS_KEEP_TIMING|SS_TICKER
+	wait = 5
+	runlevels = RUNLEVEL_GAME|RUNLEVEL_POSTGAME
+	priority = FIRE_PRIORITY_ZFALL
+	var/list/atom/movable/openspace_inhabitants = list()
+
+/datum/controller/subsystem/zfall/proc/add_openspace_inhabitant(atom/movable/A)
+	if(!istype(A) || (A in openspace_inhabitants) || (!isobj(A) && !ismob(A)))
+		return
+	RegisterSignal(A, COMSIG_PARENT_QDELETING, .proc/remove_openspace_inhabitant, A)
+	openspace_inhabitants |= A
+
+/datum/controller/subsystem/zfall/proc/remove_openspace_inhabitant(atom/movable/A)
+	SIGNAL_HANDLER
+	if(!(A in openspace_inhabitants))
+		return
+	UnregisterSignal(A, COMSIG_PARENT_QDELETING)
+	openspace_inhabitants -= A
+
+/datum/controller/subsystem/zfall/fire(resumed)
+	var/list/removals = list()
+	for(var/atom/movable/A as() in openspace_inhabitants)
+		if(MC_TICK_CHECK)
+			break
+		var/turf/T = get_turf(A)
+		if(!isopenspace(T))
+			removals |= A
+			continue
+		if(!A.zfalling)
+			T.zFall(A)
+	if(MC_TICK_CHECK)
+		return
+	for(var/A as() in removals)
+		remove_openspace_inhabitant(A)
```

Comment:
This will not work. Do what the other subsystems do

Reply:
Also why do we need a subsystem for this? This will add a delay between entering a tile and falling down. Can't we just do it on move?

Reply:
I do not know what the other subsystems do

Reply:
It already runs on move, this is needed so that if you change your "fallability" state, it will update without requiring every possible thing that could change to trigger a zfall.
For example: Removing a catwalk (That doesn't change turfs)
Turning off a jetpack
Losing your wings